src/content/blog/content-layer-deep-dive.mdx

```diff
@@ -0,0 +1,230 @@
+
+In Astro 2 we introduced [Content Collections](https://docs.astro.build/en/guides/content-collections/) as a powerful new way organize your local content, and use it to easily build sites with type-safe data. The developer experience is best in class, and it has helped developers to scale their sites to thousands of pages. However, while Content Collections are great for local files such as Markdown and MDX, we heard from you that you wanted more flexibility for that content, and you wanted the same benefits for all of your content including remote APIs. It was also clear that while many people were building sites with thousands of pages, Content Collections struggled to scale into the tens of thousands of pages, with slower builds and excessive memory usage.
+
+In June [we previewed our plan](https://astro.build/blog/future-of-astro-content-layer/) to solve these problems and more with an all new type of content collection. The Content Layer API introduces loaders, which take collections beyond files in `src/content`, letting you load your content from anywhere – whether local files on disk or remote APIs. Since its first experimental release in Astro 4.14, we've had some great feedback and have been working hard on stabilizing the API and getting it ready for release in Astro 5.
+
+## What is the Content Layer
+
+The Content Layer API allows you to load data from any source when you build your site, and then access it on your page with a simple, type-safe API. With existing content collections this is limited to files such as Markdown, inside folders in `src/content`, but the Content Layer API lets you load data from anywhere. Astro caches the data locally between builds, meaning that updates can be fast and can minimize the number of API calls that need to be made. You can combine data from many sources on each page, using the same `getEntry` and `getCollection` APIs for each. Everything is type-safe, with TypeScript types auto-generated from your schema.
+
+A **collection** is defined as a set of entries that share a common schema. Each entry has a unique ID. It is analogous to a table in a relational database. When using the Astro Content Layer API, each collection uses a **loader**, which defines how the entries are loaded to populate that collection. A loader can be a basic inline function that returns an array of entries, or it can be a more advanced object that handles its own caching and data store, and may be distributed as a module on NPM. When a site is built, the loader for each collection is invoked, which will then update the local **data store**. The Astro pages can then query that data store using the `getCollection` and `getEntry` functions. This can be done during pre-rendering, or during server rendering if using an on-demand adapter. In each of these cases the same data is available, which is the snapshot from the time of the build.
```

Comment:
```suggestion
If you've used content collections before, you'll recognize many of the following concepts and terms. In fact, we haven't changed much about *using* collections in your project at all! A **collection** is still our term for a set of entries that share a common schema. Each entry has a unique ID. It is analogous to a table in a relational database.

But now, each collection uses a **loader**, which defines how the entries are loaded to populate that collection. A loader can be a basic inline function that returns an array of entries, or it can be a more advanced object that handles its own caching and data store. The very first loaders are already distributed as modules on NPM.

Whenever your site is built, the loader for each collection is invoked which then updates the local **data store**. You can query that data store using the familiar **`getCollection()`** and **`getEntry()`** functions. This is done at build time for your prerendered pages, or during server rendering if using an on-demand adapter. In each of these cases the same data is available, which is the snapshot from the time of the build.
```